client/src/main/java/com/battleships/connection/Connection.java

```diff
@@ -0,0 +1,209 @@
+package com.battleships.connection;
+
+import com.battleships.LogMessages;
+import com.battleships.commands.CommandType;
+import com.battleships.commands.Message;
+import com.battleships.connection.commands.AbstractServerCommand;
+import com.battleships.connection.commands.ServerCommandsFactory;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+
+/**
+ *
+ */
+public enum Connection {
+    INSTANCE;
+    private Socket socket;
+    private ServerIO serverIO;
+    private final static Logger logger = LogManager.getLogger(Connection.class);
+    private Thread readCommandsFromUserThread;
+    private static final int initialConnectingTimeout = 2000;
+    private final BooleanProperty connected = new SimpleBooleanProperty(false);
+    private final BooleanProperty playerActive = new SimpleBooleanProperty(false);
+    private final BooleanProperty playerReadyProperty = new SimpleBooleanProperty(false);
+
+    public boolean getPlayerActive() {
+        return playerActive.get();
+    }
+
+    public BooleanProperty playerActiveProperty() {
+        return playerActive;
+    }
+
+    public void setPlayerActive(boolean playerActive) {
+        this.playerActive.set(playerActive);
+    }
+
+    public void setPlayerReady(boolean playerReady) {
+        this.playerReadyProperty.setValue(playerReady);
+    }
+
+    public BooleanProperty playerReadyProperty() {
+        return this.playerReadyProperty;
+    }
+
+    /**
+     * Establishes the server's connection
+     *
+     * @param connectionInfo - <code>ConnectionInfo</code> object that includes the ip and port of the server
+     */
+
+    public void establishConnection(ConnectionInfo connectionInfo) {
+        if (!isConnected()) {
+            tryToEstablishConnection(connectionInfo);
+        }
+    }
+
+    public BooleanProperty connectedProperty() {
+        return connected;
+    }
+
+    private void setConnected(boolean connected) {
+        this.connected.set(connected);
+    }
+
+    public boolean isConnected() {
+        checkConnected();
+        return connected.get();
+    }
+
+    private void checkConnected() {
+        setConnected(socket != null && socket.isConnected());
+    }
+
+    private void tryToEstablishConnection(ConnectionInfo connectionInfo) {
+        try {
+            socket = new Socket();
+            InetSocketAddress endpoint = new InetSocketAddress(connectionInfo.getIp(), connectionInfo.getPort());
```

Comment:
Całość Try i całość catch można wrzucić do osobnych metod, dzięki temu poprawia się czytelność kodu. :)

Reply:
nie chcemy tego ekstrahować, ponieważ będzie to nieczytelne, dużo jednolinijkowych metod, które odsyłają Cie do kolejnych ... :/